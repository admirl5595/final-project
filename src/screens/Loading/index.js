import React from "react";
import { View, Text, ActivityIndicator } from "react-native";
import styles from "./styles";

export default function LoadingScreen() {
  return (
    <View style={styles.container}>
      <Text style={styles.header}>Loading</Text>
      <ActivityIndicator size="large" />
    </View>
  );
}
