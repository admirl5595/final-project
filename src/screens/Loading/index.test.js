import React from "react";
import renderer from "react-test-renderer";
import { ActivityIndicator } from "react-native";

// functions for simulating user interaction
import { render, fireEvent } from "@testing-library/react-native";

// import component to be tested
import Loading from "./index";

// mock hooks from react native navigation
jest.mock("@react-navigation/native", () => {
  const actualNav = jest.requireActual("@react-navigation/native");
  return {
    ...actualNav,
    useNavigation: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
    }),
    useRoute: () => ({
      navigate: jest.fn(),
      dispatch: jest.fn(),
      params: { patient: 1 },
    }),
  };
});

// mock hooks from react native navigation
jest.mock("../../../firebase-config", () => {
  const actualConfig = jest.requireActual("../../../firebase-config");
  return {
    ...actualConfig,
    auth: { currentUser: { displayName: "bob" } },
  };
});

describe("check contents", () => {
  it("contains loading text", () => {
    const { getByText } = render(<Loading />);

    const text = getByText("Loading");

    expect(text).toBeTruthy();
  });

  it("contains loading spinner", () => {
    const tree = renderer.create(<ActivityIndicator />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
