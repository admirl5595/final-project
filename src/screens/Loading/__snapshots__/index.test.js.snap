// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check contents contains loading spinner 1`] = `<ActivityIndicator />`;
